src/Workspaces/Remote/ServiceHub/Host/RemoteSolutionCache.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote;
+
+/// <summary>
+/// LRU cache of checksum+solution pairs.  Used to keep track of the last few solutions the remote server knows about,
+/// helping to avoid unnecessary syncs/recreations of those solutions while many requests are coming into the server.
+/// Not threadsafe.  Only use while under a lock.
+/// </summary>
+internal sealed class RemoteSolutionCache<TChecksum, TSolution>
```

Comment:
><TChecksum, TSolution> [](http://example.com/codeflow?start=41&length=22)

Could consider just using the types directly in here.

Reply:
so the benefit here was that i could test the LRU behavior without acutally having to make checksum/solution instances :)  i could use any key/value in the tests (int/string for example).

Reply:
nm, I see this makes testing nicer. Maybe the name and checksum/solution references are too specific then

Reply:
fair.  i can see about tweaking them.  it is oogy :)